### PR TITLE
update logic and tests for single past country

### PR DESCRIPTION
**Pull request by raaidbroad**
## Why
Looks like there is a small bug in what country is pulled if there is only one other country of residence prior to the current residence (see relevant ticket).

[Relevant ticket](https://broadinstitute.atlassian.net/browse/DSPDC-1521?atlOrigin=eyJpIjoiZDFlM2Y3ZDUwOWYxNDhmYmI0NGMyOWIxYTZmYTkyMzMiLCJwIjoiaiJ9)

## This PR
Attempts to fix said issue and update tests to enforce correct behavior.

### CODE REVIEW

**Review by aherbst-broad — APPROVED** [review]
This looks ok, @quazi-broad should take a look to ensure the business logic is sound.

File: hles/transformation/src/main/scala/org/broadinstitute/monster/dap/dog/ResidentialEnvironmentTransformations.scala [hunk 1]
```diff
@@ -65,9 +65,9 @@ object ResidentialEnvironmentTransformations {
           // there are two separate fields to check for "First country"
           dePastResidenceCountry1 =
             if (pastCountryCount > 1)
```
Comment (ThePletch) [inline hles/transformation/src/main/scala/org/broadinstitute/monster/dap/dog/ResidentialEnvironmentTransformations.scala:67]: it won't let me make a suggestion that spans deleted lines, so don't click auto-accept, but this is my suggestion for simplifying this field's logic

```suggestion
          dePastResidenceCountry1 =
            if (pastCountryCount == 1) rawRecord.getOptional("de_country_01_only_dd") else getPastResidenceCountry(rawRecord, pastCountryCount, 1L, "de_country_01_dd")
```

File: hles/transformation/src/main/scala/org/broadinstitute/monster/dap/dog/ResidentialEnvironmentTransformations.scala [hunk 2]
```diff
@@ -92,7 +92,8 @@ object ResidentialEnvironmentTransformations {
...
               rawRecord
                 .getOptional("de_country_01")
-                .orElse(rawRecord.getOptional("de_country_01_only"))
+            else if (pastCountryCount == 1)
+              rawRecord.getOptional("de_country_01_only")
```
Comment (ThePletch) [inline hles/transformation/src/main/scala/org/broadinstitute/monster/dap/dog/ResidentialEnvironmentTransformations.scala:96]: see above - i think we can simplify this block in the same way

**Review by ThePletch — APPROVED** [review]
logic is correct, but one suggestion to improve readability/brevity

**Review by aherbst-broad — APPROVED** [review]

**Pull request merged**